picasso/src/main/java/com/squareup/picasso3/BitmapUtils.java

```diff
@@ -75,28 +75,32 @@ static boolean shouldResize(boolean onlyScaleDown, int inWidth, int inHeight,
         || (targetHeight != 0 && inHeight > targetHeight);
   }
 
-  static void calculateInSampleSize(int reqWidth, int reqHeight, int width, int height,
+  static void calculateInSampleSize(int requestWidth, int requestHeight, int width, int height,
       BitmapFactory.Options options, Request request) {
-    int sampleSize = 1;
-    if (height > reqHeight || width > reqWidth) {
-      final int heightRatio;
-      final int widthRatio;
-      if (reqHeight == 0) {
-        sampleSize = (int) Math.floor((float) width / (float) reqWidth);
-      } else if (reqWidth == 0) {
-        sampleSize = (int) Math.floor((float) height / (float) reqHeight);
-      } else {
-        heightRatio = (int) Math.floor((float) height / (float) reqHeight);
-        widthRatio = (int) Math.floor((float) width / (float) reqWidth);
-        sampleSize = request.centerInside
-            ? Math.max(heightRatio, widthRatio)
-            : Math.min(heightRatio, widthRatio);
-      }
-    }
-    options.inSampleSize = sampleSize;
+    options.inSampleSize = ratio(requestWidth, requestHeight, width, height, request);
     options.inJustDecodeBounds = false;
   }
 
+  static int ratio(int requestWidth, int requestHeight, int width, int height, Request request) {
+    if (height <= requestHeight && width <= requestWidth) {
+      return 1;
+    }
+    if (requestHeight == 0) {
+      int widthRatio = width / requestWidth;
```

Comment:
You can move this above the `if` since we need to compute it no matter what.

Reply:
if requestHeight != 0 and requestWidth == 0, then we don't need to compute widthRatio, though.

Reply:
True. Let's leave it as is.